src/lib/Hydra/Plugin/RunCommand.pm

```diff
@@ -160,9 +163,22 @@ sub buildFinished {
 
         $runlog->started();
 
-        system("$command") == 0
+        my $logPath = Hydra::Helper::Nix::constructRunCommandLogPath($runlog) or die "RunCommandLog not found.";
+        my $dir = dirname($logPath);
+        my $oldUmask = umask();
+
+        # file: 640, dir: 750
+        umask(0027);
+        make_path($dir);
+
+        open(my $f, '>', $logPath);
+        umask($oldUmask);
```

Comment:
If we get a die or whatever in make_path / open / etc. we could leave this code path and enter another without resetting the umask. That could create some bizarre bug reports.

How about we wrap this up with Try::Tiny:

```
```
```suggestion
        my $oldUmask = umask();
        try {
            # file: 640, dir: 750
            umask(0027);
            make_path($dir);

            open(my $f, '>', $logPath);
        } finally {
            umask($oldUmask);
        }
```

Reply:
Should the entire  `run3` logic be moved into the try block as well (including the later `close`)?